Source/JavaScriptCore/API/JSContext.mm

```diff
@@ -265,7 +265,9 @@ - (NSString *)name
     if (!name)
         return nil;
 
-    return adoptCF(JSStringCopyCFString(kCFAllocatorDefault, name)).bridgingAutorelease();
+    NSString *result = adoptCF(JSStringCopyCFString(kCFAllocatorDefault, name)).bridgingAutorelease();
+    JSStringRelease(name);
+    return result;
```

Comment:
Is it correct? Now, ownership gets moved from CF to NSString, and it is returning that, right?

Reply:
Yes, this is correct. `.bridgingAutorelease()` will turn it to an auto release object. That means that it will be released automatically at a later point in time (typically when the current autorelease pool drains). So, the object in auto release mode will be returned. I have tested it locally with `leaks` and `[result release];`. I believe it's correct.

Reply:
Ah, I see. `name` is leaked. The result is not.

Reply:
How about using `JSRetainPtr`? It makes code safer.

```
auto name = adopt(JSGlobalContextCopyName(m_context));
if (!name)
    return nil;

return adoptCF(JSStringCopyCFString(kCFAllocatorDefault, name.get())).bridgingAutorelease();
```

Reply:
Sweet, even better. Done.